src/ngram_matches.cc

```diff
@@ -44,14 +44,13 @@ namespace fuzzy
     return _psentences.size();
   }
 
-  std::vector<unsigned>
-  NGramMatches::sentence(int s_id) const
+  const unsigned*
+  NGramMatches::sentence(unsigned s_id, unsigned* length) const
   {
     unsigned idx = _suffixArray[s_id];
```

Comment:
Automatic type deduction should be fine here, I suggest replacing type with `const auto`, and line below with `const auto*`.